### PR TITLE
ref(swagger):add /v2 endpoints to support existing clients

### PR DESCRIPTION
**Pull request by kmala**

### CODE REVIEW

**Comment by krancour** [comment]
What are the existing clients we are trying to support by doing this?

**Comment by kmala** [comment]
all the beta and rc workflow-manger containers

**Comment by krancour** [comment]
I wouldn't think it's necessary to provide backward compatibility with anything that is pre-GA.

**Comment by kmala** [comment]
yes...i agree with you but this is special scenario for us as we want to know the information of what people are running currently especially how many are in beta and are upgrading to stable

**Comment by krancour** [comment]
> this is special scenario for us as we want to know the information of what people are running currently

Right. I forgot about that.  But isn't that more relevant for whatever centralized component we host for wfm to talk back to?

File: api/swagger-spec/swagger.yml [hunk 1]
```diff
@@ -178,6 +177,56 @@ paths:
           description: unexpected error
           schema:
             $ref: "#/definitions/error"
+  /v2/versions/latest:
+    post:
+      operationId: getComponentsByLatestReleaseForV2
+      summary: "list the latest release version of the components.This endpoint is to support existing clients"
```
Comment (arschles) [inline api/swagger-spec/swagger.yml]: this should say "This endpoint is to support _old_ clients"

File: api/swagger-spec/swagger.yml [hunk 2]
```diff
@@ -178,6 +177,56 @@ paths:
...
+                items:
+                  $ref: "#/definitions/componentVersion"
+        default:
+          description: unexpected error
+          schema:
+            $ref: "#/definitions/error"
+  /v2/clusters/{id}:
+    post:
+      operationId: createClusterDetailsForV2
+      summary: "create a cluster with all components.This endpoint is to support existing clients"
```
Comment (arschles) [inline api/swagger-spec/swagger.yml]: same here regarding "old clients" language

**Comment by arschles** [comment]
@krancour yes, you're right. the versions server should provide full backward compatibility, and ironically that's what this PR works toward. the swagger document here is the API definition shared by the client and the server. the workflow-manager client code will use the `v3` endpoints but this change will allow the server to handle old clients that use the `v2` endpoints

**Pull request merged**